lang/cs/Org.Apache.REEF.IMRU/API/IMRUJobDefinition.cs

```diff
@@ -27,23 +27,139 @@ namespace Org.Apache.REEF.IMRU.API
     /// <seealso cref="IMRUJobDefinitionBuilder" />
     public sealed class IMRUJobDefinition
     {
-        private readonly IConfiguration _configuration;
         private readonly string _jobName;
+        private readonly IConfiguration _mapFunctionConfiguration;
+        private readonly IConfiguration _mapInputCodecConfiguration;
+        private readonly IConfiguration _updateFunctionCodecsConfiguration;
+        private readonly IConfiguration _reduceFunctionConfiguration;
+        private readonly IConfiguration _updateFunctionConfiguration;
+        private readonly IConfiguration _mapOutputPipelineDataConverterConfiguration;
+        private readonly IConfiguration _mapInputPipelineDataConverterConfiguration;
+        private readonly IConfiguration _partitionedDatasetConfiguration;
+        private readonly int _numberOfMappers;
 
-        internal IMRUJobDefinition(IConfiguration configuration, string jobName)
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="mapFunctionConfiguration">Map function configuration</param>
+        /// <param name="mapInputCodecConfiguration">Map input codec configuration</param>
+        /// <param name="updateFunctionCodecsConfiguration">codec configuration for update 
+        /// function. It is union of TMapInput, TMapOutput and TResult configuration</param>
+        /// <param name="reduceFunctionConfiguration">Reduce function configuration</param>
+        /// <param name="updateFunctionConfiguration">Update function configuration</param>
+        /// <param name="mapOutputPipelineDataConverterConfiguration">Configuration of 
+        /// PipelineDataConverter for TMapOutput</param>
+        /// <param name="mapInputPipelineDataConverterConfiguration">Configuration of 
+        /// PipelineDataConverter for TMapInput</param>
+        /// <param name="partitionedDatasetConfiguration">Configuration of partitioned 
+        /// dataset</param>
+        /// <param name="numberOfMappers">Number of mappers</param>
+        /// <param name="jobName">Job name</param>
+        internal IMRUJobDefinition(
```

Comment:
Markus,

Just to be sure if I have MapInput which is nothing but say consists of two
floats and hence in its MapInput codec config say config1  I would have
binded IStreamingCodec<float> to say FloatCodec as base codec and I have
MapOutput which is same as MapOutput and in its  codec config say config2 I
have again   binded IStreamingCodec<float> to FloatCodec, and then I do
Configurations.Merge(config1,config2), it will not throw me the error?

On Tue, Aug 4, 2015 at 5:11 PM, Markus Weimer <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.IMRU/API/IMRUJobDefinition.cs
> <https://github.com/apache/incubator-reef/pull/334#discussion_r36257723>:
>
> > +        /// </summary>
> > +        /// <param name="mapFunctionConfiguration">Map function configuration</param>
> > +        /// <param name="mapInputCodecConfiguration">Map input codec configuration</param>
> > +        /// <param name="updateFunctionCodecsConfiguration">codec configuration for update
> > +        /// function. It is union of TMapInput, TMapOutput and TResult configuration</param>
> > +        /// <param name="reduceFunctionConfiguration">Reduce function configuration</param>
> > +        /// <param name="updateFunctionConfiguration">Update function configuration</param>
> > +        /// <param name="mapOutputPipelineDataConverterConfiguration">Configuration of
> > +        /// PipelineDataConverter for TMapOutput</param>
> > +        /// <param name="mapInputPipelineDataConverterConfiguration">Configuration of
> > +        /// PipelineDataConverter for TMapInput</param>
> > +        /// <param name="partitionedDatasetConfiguration">Configuration of partitioned
> > +        /// dataset</param>
> > +        /// <param name="numberOfMappers">Number of mappers</param>
> > +        /// <param name="jobName">Job name</param>
> > +        internal IMRUJobDefinition(
>
> I don't think so. As long as we bind the same implementation to both named
> parameters, we should be good to go.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/334/files#r36257723>.
>